### PR TITLE
Add has_$option

### PR DESCRIPTION
**Pull request by kiwiroy**
Make *options* a little more *Moose*y even when class isn't Moose derived.

Up for discussion, needs additions to documentation at least.

## References

* #11

### CODE REVIEW

File: lib/Applify.pm [hunk 1]
```diff
@@ -321,12 +321,14 @@ sub _generate_application_class {
...
       my $name = $option->{name};
-      my $fqn = join '::', $application_class, $name;
       if ($meta) {
         $meta->add_attribute($name => {is => 'rw', default => $option->{default}});
       }
       else {
+        my $fqn = join '::', $application_class, $name;
         _sub($fqn => sub { @_ == 2 and $_[0]->{$name} = $_[1]; $_[0]->{$name} });
+        $fqn = join '::', $application_class, join '_', has => $name;
+        _sub($fqn => sub { !!(exists $_[0]->{$name} && defined $_[0]->{$name}) });
```
Comment (jhthorsen) [inline lib/Applify.pm]: Why do you need both exists and defined? I'm guessing one of them should be enough, though I cannot decide on which is the best option.

File: t/options.t [hunk 2]
```diff
@@ -53,4 +53,33 @@ eval {
...
+app { };
+HERE
+
+$script = $app->_script;
+
+my $instance = app_instance($script, qw{-input-file /tmp/test});
+is $instance->has_input_file, 1, 'Moose style';
+is !$instance->has_iii, 1, 'does not exist';
+is $instance->has_output_file, 1, 'default always exists';
+is $instance->has_template, 0, 'has_template not replaced see _sub()';
```
Comment (jhthorsen) [inline t/options.t]: I'm a lot more positive to the change, since it won't replace any existing methods.

Great work! 👍

**Comment by jhthorsen** [comment]
Do you know why it says "This pull request is still a work in progress"? Is that something you set or do I have to do something?

File: lib/Applify.pm (hunk 1 above)
Comment (kiwiroy) [reply lib/Applify.pm]: `defined` as every one gets a *default*
https://github.com/jhthorsen/applify/blob/df2b22ba4ea09a34e4745332ebbe4775496887b4/lib/Applify.pm#L130

**Comment by kiwiroy** [comment]
@jhthorsen I put it in draft mode, have fixed the sub and added information about the change to the pod.

**Comment by jhthorsen** [comment]
Ok. Gonna merge it then :)

**Pull request merged**